src/openms_gui/source/VISUAL/SpectraTreeTab.cpp

```diff
@@ -306,6 +313,7 @@ namespace OpenMS
       clear();
       return;
     }
+    layer_ = layer;
```

Comment:
member can be removed IMHO

Reply:
it is used in   void SpectraTreeTab::updateIndexFromCurrentLayer()
  {
    spectra_treewidget_->setTreePosition(layer_->getCurrentSpectrumIndex());
  }